src/frontends/tensorflow/src/op/pad.cpp

```diff
@@ -3,71 +3,80 @@
+    auto unsqueeze_pad_begin = make_shared<Gather>(paddings, index_zero, axis);
+    auto unsqueeze_pad_end = make_shared<Gather>(paddings, index_one, axis);
+
+    pads_begin = make_shared<Squeeze>(unsqueeze_pad_begin, axis);
+    pads_end = make_shared<Squeeze>(unsqueeze_pad_end, axis);
+}
+
+static OutputVector translate_pad_base_op(const NodeContext& node,
+                                          const Output<Node>& input,
+                                          const Output<Node>& paddings,
+                                          const Output<Node>& constant_value) {
+    auto pad_mode = ov::op::PadMode::CONSTANT;
+
+    // prepare pads_begin and pads_end for OpenVINO Pad
+    shared_ptr<Node> pads_begin, pads_end;
+    slice_pads_begin_end(paddings, pads_begin, pads_end);
+
+    auto pad = make_shared<Pad>(input, pads_begin, pads_end, constant_value, pad_mode);
+    set_node_name(node.get_name(), pad);
+    return {pad};
+}
 
 OutputVector translate_pad_op(const NodeContext& node) {
-    auto ng_input = node.get_input(0), ng_paddings_op = node.get_input(1);
-    Output<Node> pad_val_op;
+    default_op_checks(node, 2, {"Pad"});
+    auto input = node.get_input(0);
+    auto paddings = node.get_input(1);
+    auto constant_value = make_shared<Constant>(input.get_element_type(), Shape{}, 0);
 
-    // Set inputs and pad_val_op
-    auto op_type = node.get_op_type();
-    if (op_type == "Pad" || op_type == "MirrorPad") {
-        pad_val_op = make_shared<Constant>(ng_input.get_element_type(), Shape(), std::vector<int>({0}));
-    } else if (op_type == "PadV2") {
-        pad_val_op = node.get_input(2);
-    } else {
-        TENSORFLOW_OP_VALIDATION(node, false, "Incorrect TF Pad OpType: " + node.get_op_type());
-    }
+    return translate_pad_base_op(node, input, paddings, constant_value);
+}
 
-    // Set pad_mode
-    auto pad_mode = ov::op::PadMode::CONSTANT;
-    if (op_type == "MirrorPad") {
-        auto pad_mode_str = node.get_attribute<std::string>("mode");
-        if (pad_mode_str == "REFLECT") {
-            pad_mode = ov::op::PadMode::REFLECT;
-        } else if (pad_mode_str == "SYMMETRIC") {
-            pad_mode = ov::op::PadMode::SYMMETRIC;
-        } else {
-            TENSORFLOW_OP_VALIDATION(node, false, pad_mode_str + " is not an allowed padding mode.");
-        }
-    }
+OutputVector translate_padv2_op(const NodeContext& node) {
+    default_op_checks(node, 3, {"PadV2"});
+    auto input = node.get_input(0);
+    auto paddings = node.get_input(1);
+    auto constant_value = node.get_input(2);
+
+    return translate_pad_base_op(node, input, paddings, constant_value);
+}
+
+OutputVector translate_mirror_pad_op(const NodeContext& node) {
+    default_op_checks(node, 2, {"MirrorPad"});
+    auto input = node.get_input(0);
+    auto paddings = node.get_input(1);
+
+    // retrieve attributes
+    auto mode = node.get_attribute<std::string>("mode");
+    auto pad_mode = convert_padding_mode(node, mode);
```

Comment:
Please add additional attribute `ov::op::PadMode pad_mode` to`translate_pad_base_op` function, after that, I believe we will be able to use this function here.
for Pad, Padv2 we will pass ov::op::PadMode::CONSTANT value
for MirrorPad we will pass this converted padding mode

Reply:
Not sure, how to do this I double-checked. Let us discuss this offline.

Reply:
Will discuss and fix in the next PR if needed.

Reply:
In any case there will branching in creation of `Pad` operation because for `Pad` and `PadV2` we also use constant value. These types of padding are a bit different and makes sense to differentiate them. Better not to mix-up